src/main/java/org/intellij/sdk/codesync/Constants.java

```diff
@@ -226,4 +226,18 @@ private ErrorCodes() {
         public static final int DIFFS_LIMIT_REACHED = 402;
         public static final int ACCOUNT_DEACTIVATED = 403;
     }
+
+    public static final class CustomErrorCodes {
+        /*
+            Custom error codes for the case: Free Plan restrictions
+        */
+        private CustomErrorCodes(){
+            // restrict instantiation
+        }
+
+        public static final int IS_FROZEN_REPO = 4000;
+        public static final int REPO_SIZE_LIMIT_REACHED = 4001;
```

Comment:
I think we this PR targets ```PRIVATE_REPO_COUNT_LIMIT_REACHED = 4006``` error code instead of these

Reply:
I think above comment still needs to be addressed @saadberry

Reply:
I had kept them for future cases but i have realized there will be no case to send 4001 from plugin. I shall remove it then. But there can be a case for sending 4000, in case a user tries to modify a frozen repo right? So we should keep that